### PR TITLE
Add Sensei bundle settings

### PR DESCRIPTION
**Pull request by renatho**
Related to https://github.com/Automattic/themes/issues/7498

Based off of https://github.com/Automattic/wp-calypso/pull/84886

## Proposed Changes

* It adds the Sensei bundle settings.

## Testing Instructions

* Use it on your sandbox: D131017-code.

**Getting theme in an existing site with free plan**

* Navigate to `http://calypso.localhost:3000/start`.
* Get a free plan, and pick a free theme.
* After the site is created, navigate to Appearance > Themes.
* Search for "Course" theme, and install it.
* Check that the flow displays the Sensei information where needed, it asks a plan upgrade, auto-installs Sensei LMS plugin, and redirects you to the Sensei Setup Wizard.

**Getting theme through the design picker**

* Navigate to `http://calypso.localhost:3000/start`.
* Get a free plan, and pick any theme.
* In the URL, update the `theme` query string to receive `course`.
* Check that it displays the Sensei information properly, and continue through the flow.
* Check that the flow displays the Sensei information where needed, it asks a plan upgrade, auto-installs Sensei LMS plugin, and redirects you to the Sensei Setup Wizard.

## Screenshots

See the badge and the tooltip:

Screenshot 2023-12-14 at 14 36 25

See the banner text:

Screenshot 2023-12-14 at 14 36 39

Badge and tooltip:
Screenshot 2023-12-18 at 14 39 51

Icon and text:
Screenshot 2023-12-18 at 13 28 36

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/theme/hooks/use-bundle-settings.tsx
Comment (renatho) [inline client/my-sites/theme/hooks/use-bundle-settings.tsx:86]: I used the same from WooCommerce for almost all the texts, and just changed the name to "Sensei". The only one that is a little different is this one, which I copied the text we have on senseilms.com home.

File: client/landing/stepper/declarative-flow/plugin-bundle-data.ts
Comment (renatho) [inline client/landing/stepper/declarative-flow/plugin-bundle-data.ts]: When the user installs a theme with Sensei LMS bundled, they will be redirected to the Sensei Setup Wizard after the WPCOM flow is complete.

**Comment by dsas** [comment]
Not a complete test, but you can partially test onboarding (/start). When you get to the theme step choose a theme and then change the `theme=` GET parameter in the url. 
Screenshot 2023-12-18 at 14 39 51

File: client/my-sites/theme/hooks/use-bundle-settings.tsx
Comment (dsas) [inline client/my-sites/theme/hooks/use-bundle-settings.tsx]: This sentence is missing a `the` (as is the Woo version). Surely it should read "This theme comes bundled with the Sensei plugin"
Screenshot 2023-12-18 at 14 50 19

File: client/landing/stepper/declarative-flow/plugin-bundle-data.ts
Comment (dsas) [inline client/landing/stepper/declarative-flow/plugin-bundle-data.ts:102]: Completely a different problem for a different pr and repo, but the setup wizard encourages you to use the Course theme even if you already have the Course theme activated.

Comment (renatho) [reply client/landing/stepper/declarative-flow/plugin-bundle-data.ts:102]: Good catch, @dsas!
Sorry that I didn't mention it previously in this PR, but it's already in our backlog to be changed in the next few days! https://github.com/Automattic/sensei/issues/6350

File: client/my-sites/theme/hooks/use-bundle-settings.tsx
Comment (renatho) [reply client/my-sites/theme/hooks/use-bundle-settings.tsx]: Updated for Woo and for Sensei in this commit: https://github.com/Automattic/wp-calypso/pull/85309/commits/aaf49ce0dfbe7caf4038be97f197210e77afc4db 😉

**Comment by renatho** [comment]
> Not a complete test, but you can partially test onboarding (/start). When you get to the theme step choose a theme and then change the theme= GET parameter in the url.

Thank you, @dsas!
I updated the screenshots and the test instructions to have this case.

**Review by dsas — APPROVED** [review]
Tested with the sandbox and it wfm

**Comment by renatho** [comment]
Thank you for all the reviews, @dsas!

Just to confirm, when you tested it did you also test that WooCommerce bundle continued working with the sandbox diff? D131017#2569938-code

I'll wait for your confirmation, and for the E2E tests fix to make sure I'm not introducing any new bugs. After that, I'll deploy everything! 🙌🙌🙌

**Comment by dsas** [comment]
Sorry I missed that comment on phab somehow, yeah I tested:
 * Woo with D131017 & without this pr
 * Woo with D131017 & with this PR
 * Sensei with D131017 & with this PR

**Comment by renatho** [comment]
Sorry for the noise! It automatically added a few teams as reviewers after an accidental wrong merge in this branch. It was already reverted.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10449591

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @renatho for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10449591).

**Pull request merged**
